pygmt/modules.py

```diff
@@ -214,3 +217,74 @@ def __exit__(self, exc_type, exc_value, traceback):
         )
         with Session() as lib:
             lib.call_module("set", arg_str)
+
+
+@xr.register_dataarray_accessor("gmt")
+class GMTDataArrayAccessor:
+    """
+    This is the GMT extension for :class:`xarray.DataArray`.
+
+    You can access various GMT specific metadata about your grid as follows:
+
+    >>> from pygmt.datasets import load_earth_relief
+    >>> # Use the global Earth relief grid with 1 degree spacing
+    >>> grid = load_earth_relief(resolution="01d")
+
+    >>> # See if grid uses Gridline (0) or Pixel (1) registration
+    >>> grid.gmt.registration
+    1
+    >>> # See if grid uses Cartesian (0) or Geographic (1) coordinate system
+    >>> grid.gmt.gtype
+    1
+    """
+
+    def __init__(self, xarray_obj):
+        self._obj = xarray_obj
+        try:
+            self._source = self._obj.encoding["source"]
+            self._registration, self._gtype = map(
+                int, grdinfo(self._source, C="n", o="10,11").split()
+            )
+        except KeyError:
+            self._registration = 0  # Default to Gridline registration
+            self._gtype = 0  # Default to Cartesian grid type
+            # logging.warning(
+            #     msg="Cannot find a NetCDF source for the xarray grid. "
+            #     "Will fallback to using GMT's default setting to assume "
+            #     "'Gridline node registration used [Cartesian grid]'"
+            # )
+
+    @property
+    def registration(self):
+        """
+        Registration type of the grid, either Gridline (0) or Pixel (1).
+        """
+        return self._registration
+
+    @registration.setter
+    def registration(self, value):
+        if value in (0, 1):
```

Comment:
It may be more user friendly if we can also set the properties by `grid.gmt.registration = "pixel"` and `grid.gmt.gtype = "geographic"`, but I'm OK with the current boolean values, 0 and 1.

Reply:
Good idea. Let's leave it for now, but introduce this once we work out how to do the `__repr__` as mentioned in https://github.com/GenericMappingTools/pygmt/pull/500#pullrequestreview-446833212.